Tools/autotest/arducopter.py

```diff
@@ -12691,7 +12691,112 @@ def ScriptMountPOI(self):
 
         self.context_pop()
         self.reboot_sitl()
+    
+    def ScriptMountAllModes(self):
+        '''test location from all mount modes using the scripting backend'''
+        self.context_push()
+        
+        # enable scripting and set mount type to scripting
+        self.set_parameters({
+            "SCR_ENABLE": 1,
+            "MNT1_TYPE": 9,
+        })    
+        self.reboot_sitl()
+        
+        # install get-target-location-mount-backend.lua script
+        self.install_example_script_context('get-target-location-mount-backend.lua')
+        self.reboot_sitl()
+        
+        self.wait_ready_to_arm()
+        self.arm_vehicle()
+        self.takeoff(50)
+        
+        # capture STATUSTEXT messages
+        self.context_collect("STATUSTEXT")
+        
+        # Test RETRACT (mode 0)
+        self.run_cmd(
+            mavutil.mavlink.MAV_CMD_DO_MOUNT_CONTROL,
+            p7=mavutil.mavlink.MAV_MOUNT_MODE_RETRACT
+        )      
+        self.wait_statustext("Mode 0: No target", check_context=True)
+        
+        # Test NEUTRAL (mode 1)
+        self.run_cmd(
+            mavutil.mavlink.MAV_CMD_DO_MOUNT_CONTROL,
+            p7=mavutil.mavlink.MAV_MOUNT_MODE_NEUTRAL
+        )
+        self.wait_statustext("Mode 1: No target", check_context=True)
+        
+        # Test MAVLINK_TARGETING (mode 2)
+        self.run_cmd(
+            mavutil.mavlink.MAV_CMD_DO_MOUNT_CONTROL,
+            p7=mavutil.mavlink.MAV_MOUNT_MODE_MAVLINK_TARGETING
+        )
+        self.wait_statustext("Mode 2: No target", check_context=True)
+        
+        # Test RC_TARGETING (mode 3)
+        self.run_cmd(
+            mavutil.mavlink.MAV_CMD_DO_MOUNT_CONTROL,
+            p7=mavutil.mavlink.MAV_MOUNT_MODE_RC_TARGETING
+        )
+        self.wait_statustext("Mode 3: No target", check_context=True)
+        
+        roi_lat = -35.363150
+        roi_lng = 149.165320
+        roi_alt = 580.0
+        
+         # Set ROI location
+        self.run_cmd(
+            mavutil.mavlink.MAV_CMD_DO_SET_ROI_LOCATION,
+            p5 = roi_lat,
+            p6 = roi_lng,
+            p7 = roi_alt
+        )
+        expected_roi_lat = f"{roi_lat:.3f}"
+        expected_roi_lng = f"{roi_lng:.3f}"
+        
+        # Test GPS_POINT (mode 4)
+        self.run_cmd(
+            mavutil.mavlink.MAV_CMD_DO_MOUNT_CONTROL,
+            p7=mavutil.mavlink.MAV_MOUNT_MODE_GPS_POINT
+        )
+        self.wait_statustext(
+            rf"Mode 4:.*{expected_roi_lat}.*{expected_roi_lng}",
+            check_context=True, 
+            regex=True
+        )
+        
+        home_loc = self.mav.location()
+        expected_lat = "%.3f" % home_loc.lat
+        expected_lng = "%.3f" % home_loc.lng
```

Comment:
Good to have some variety.

But maybe choose one way or the other to truncate these values (see line 12756)